### PR TITLE
API-300_API-301_API-302: Add create channel, upsert channel and upsert list of channels

### PR DESCRIPTION
**Pull request by momoss**

### CODE REVIEW

**Comment by momoss** [comment]
[Green CI](https://core-ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fphp-api-client/detail/PR-63/2/pipeline/)

File: src/Api/ChannelApiInterface.php [hunk 1]
```diff
@@ -9,6 +9,6 @@
  * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-interface ChannelApiInterface extends ListableResourceInterface, GettableResourceInterface
+interface ChannelApiInterface extends ListableResourceInterface, GettableResourceInterface, CreatableResourceInterface, UpsertableResourceInterface, UpsertableResourceListInterface
```
Comment (ahocquard) [inline src/Api/ChannelApiInterface.php]: split on several lines

File: tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                ],
+            ]
+        );
+
+        $this->assertSame(204, $response);
+
+        $channel = $api->get('ecommerce');
+        $this->assertSameContent(
+            [
+                'currencies'       => [
```
Comment (ahocquard) [inline tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php]: add code in the content

File: tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            [
+                'currencies'    => [
+                    'EUR',
+                    'USD',
+                ],
+                'locales'       => [
+                    'fr_FR',
+                    'en_US',
+                ],
+                'category_tree' => ['2014_collection'],
```
Comment (ahocquard) [inline tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php]: just keep this line, to see what is the wrong data quickly

File: tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    /**
+     * @expectedException \Akeneo\Pim\Exception\UnprocessableEntityHttpException
+     */
+    public function testUpsertInvalidCodeFail()
+    {
+        $api = $this->createClient()->getChannelApi();
+        $api->upsert(
+            'invalid code !',
+            [
+                'currencies'    => [
```
Comment (ahocquard) [inline tests/v1_8/Api/Channel/UpsertChannelApiIntegration.php]: same, no need to send content

**Comment by momoss** [comment]
[Green CI](https://core-ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fphp-api-client/detail/PR-63/3/pipeline/)

**Pull request merged**